include/fplus/benchmark_session.hpp

```diff
@@ -0,0 +1,359 @@
+// Copyright 2015, Tobias Hermann and the FunctionalPlus contributors.
+// https://github.com/Dobiasd/FunctionalPlus
+// Distributed under the Boost Software License, Version 1.0.
+// (See accompanying file LICENSE_1_0.txt or copy at
+//  http://www.boost.org/LICENSE_1_0.txt)
+
+#include <vector>
+#include <mutex>
+#include <fplus/fplus.hpp>
+
+namespace fplus
+{
+using FunctionName = std::string;
+struct benchmark_function_report
+{
+    size_t nb_calls;
+    ExecutionTime total_time;
+    ExecutionTime average_time;
+    ExecutionTime deviation;
+};
+
+
+namespace internal
+{
+    std::string show_benchmark_function_report(const std::map<FunctionName, benchmark_function_report> & reports);
```

Comment:
Feel free to add line breaks between function parameters (if not too ugly) to stay below 80 characters per line.